### PR TITLE
update docker helper to dynamically fetch latest CHT version from staging, use kill instead of stop

### PR DESCRIPTION
**Pull request by mrjones-plip**
# Description

# Code review checklist

- [X] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [X] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:update-release-templates-for-helper-bumps/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:update-release-templates-for-helper-bumps/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:update-release-templates-for-helper-bumps/docker-compose/cht-couchdb-clustered.yml)
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
@garethbowen  - fair point!  I respect the push back.

We actually don't have `jq` on most machines we run the helper on though.  Is the JSON structure static such that we can use a brittle parsing with `cut` and `tr` ?  This works for me now, but want to be sure it will work tomorrow:

```
curl -s https://staging.dev.medicmobile.org/_couch/builds_4/_design/builds/_view/releases\?limit\=1\&descending\=true | tr -d \\n | cut -f10 -d'"'|cut -f3 -d':'
4.0.1
```

**Comment by garethbowen** [comment]
@mrjones-plip I think it's stable enough to use. You can drop the second `cut` and use the whole ID to make it more stable:

```
curl -s https://staging.dev.medicmobile.org/_couch/builds_4/_design/builds/_view/releases\?limit\=1\&descending\=true | tr -d \\n | cut -f10 -d'"'
medic:medic:4.0.1
```

**Comment by mrjones-plip** [comment]
Ah - I think, for once in it's life, regex might actually be more stable?  Your idea to remove a `cut` reminded me about "cut"ing out the part we need with `grep`:

```
curl -s https://staging.dev.medicmobile.org/_couch/builds_4/_design/builds/_view/releases\?limit\=1\&descending\=true |  tr -d \\n | grep -o 'medic\:medic\:[0-9\.]*'  
medic:medic:4.0.1
```

I'll pull update this PR tomorrow and send it your way! Thanks again.

**Comment by mrjones-plip** [comment]
@garethbowen  - fixed in d8bd6e5e6 - can you have a go at this review?

File: scripts/docker-helper-4.x/cht-docker-compose.sh [hunk 1]
```diff
@@ -46,16 +46,20 @@ CHT_NETWORK=$projectName-cht-net
 EOL
 }
 
+get_staging_url_base() {
```
Comment (garethbowen) [inline scripts/docker-helper-4.x/cht-docker-compose.sh:49]: Naming suggestion: what about `get_latest_release_url`? I feel like it needs the "latest release" bit to really convey what it's doing.

**Review by garethbowen — APPROVED** [review]
Awesome! One optional suggestion inline...

**Comment by mrjones-plip** [comment]
@garethbowen  - can you mash the `Yo...merge anyway, forget you CI!!` button you have?  I'm on my 4th or so attempt to get CI to pass on this PR of just 1 bash script changed.

**Comment by garethbowen** [comment]
Done.

**Comment by mrjones-plip** [comment]
cc @m5r & @jkuester - we now dynamically fetch the latest CHT version and are not pinned to alpha/master

**Pull request merged**